plugins/cc/cctac.ml

```diff
@@ -310,72 +285,87 @@ let type_and_refresh c =
     Proofview.tclTHEN (Proofview.Unsafe.tclEVARS evm) (Proofview.tclUNIT ty)
   end
 
+let type_and_refresh_ env sigma c =
+  let sigma, ty = Typing.type_of env sigma c in
+  let sigma, ty = refresh_type env sigma ty in
+  sigma, ty
+
 let constr_of_term c = EConstr.of_constr (ATerm.constr c)
 
-let rec proof_tac p : unit Proofview.tactic =
+let app_global_ env sigma ref args =
```

Comment:
```suggestion
let checked_app_global env sigma ref args =
```

Reply:
I don't want to call it checked right now because it doesn't check everything, and I plan to really defer it after anyways.

Reply:
`app_global_` is pretty bad though, there's no way to tell how it's different from `app_global` without looking at the implementation

Reply:
I guess congruence already uses plenty of _, I can accept this one